articles/introduction-to-hapi.js-framework/index.md

```diff
@@ -0,0 +1,136 @@
+<!--more-->
+### What is hapi.js
+Hapi.js(derived from Http-API) is an open-source Node.js framework used to build powerful and scalable web applications. Hapi is commonly used to build Application Programming Interface servers, HTTP-proxy applications, and websites.
+Hapi.js was created by the mobile team at Walmart Labs — led by Eran Hammer to handle their traffic for events like Black Friday, which is by far one of the busiest days for online shopping in the U.S. calendar.
+Hapi was originally built using express framework before facing challenges that drove Walmart to make hapi its own stand-alone framework.
+
+### Hapi.js features
+Hapi.js comes with many unique features that enable developers to build secure, powerful, and scalable applications.
+Some of these features include:
+1. End-to-end code hygiene which helps developers write manageable, controllable, and distributable code.
+2. Secure defaults which are updated regularly. Hapi blocks error messages that may leak information or echo back exploits.
+3. Encrypted and signed cookies, secret or key rotation, and HTTP security headers all meant to enhance the security of applications.
+4. An extensive set of official plugins that are meant to replace middleware used in frameworks such as express.
+5. Integrated Authorization and Authentication Architecture which is the most comprehensive authorization API available in Node.js.
+
+### Who uses hapi.js
+Many companies use hapi.js framework for their websites and web APIs.
+Below is a shortlist of five globally recognized companies that use hapi.js framework.
+- Commercetools
+- Brainhub.
+- Beam.
+- PayPal.
+- Clinlife.
+
+### Creating a Server with hapi.js
+Creating a server using hapi.js is quite easy and obviously different from other frameworks.
+Let's see how we can create a simple server using hapi.
+
+First run
+
+```
+npm init
+```
+To create a node package with the name of your choice.
+
+Then open a new terminal and navigate to your app's folder and run
+```
+npm install @hapi/hapi
+```
+to install hapi module.
+
+Afterward, navigate to the index file on your app's folder using your favorite code editor and write the code below to create the server.
+
+```javascript
+const hapi = require('hapi')
+const server = new hapi.Server()
+
+//connect the server to port 2400 of localhost
+server.connection({
+    host: 'localhost',
+    port: '2400'
+})
+
+//start the server
+server.start(error => {
+    if (error) throw error
+    else console.log('Server running at PORT 2400');
+})
+```
+You can run your server by running node 'name of your index file' on the terminal.
+
+### Creating routes with hapi.js
+Hapi uses server.route as a method to create routes. server.route methods take an options object. The object has three properties.
```

Comment:
@kayere if you could please elaborate this couple sentences - they may be good - but as a beginner myself - I found myself kind of asking more questions to myself while reading theses few sentences. If you could break down so any beginner can understand what an option object is

Reply:
Hello @hectorkambow , I have elaborated the few sentences as requested. Please review. Thanks.